hera_cal/io.py

```diff
@@ -514,6 +514,11 @@ def write_cal(fname, gains, freqs, times, flags=None, quality=None, write_file=T
                 flag_array[j, :, :, :, i] = np.ones((Nspws, Nfreqs, Ntimes), np.bool)
                 quality_array[j, :, :, :, i] = np.ones((Nspws, Nfreqs, Ntimes), np.float)
 
+    # Check gain_array for values close to zero, if so, set to 1
+    zero_check = np.isclose(gain_array, 0)
```

Comment:
I'd set the atol and rtol to be a bit more stringent. Like rtol = 1e-8 and atol = 1e-10.